addons/binding/org.openhab.binding.plugwise/src/main/java/org/openhab/binding/plugwise/handler/PlugwiseRelayDeviceHandler.java

```diff
@@ -0,0 +1,576 @@
+            sendMessage(new ClockGetRequestMessage(macAddress));
+        }
+
+        @Override
+        public boolean shouldBeScheduled() {
+            return thing.getStatus() == ONLINE && isLinked(CHANNEL_CLOCK);
+        }
+    };
+
+    private final PlugwiseDeviceTask currentPowerUpdateTask = new PlugwiseDeviceTask("Current power update",
+            scheduler) {
+        @Override
+        public int getConfiguredInterval() {
+            return getChannelUpdateInterval(CHANNEL_POWER);
+        }
+
+        @Override
+        public void runTask() {
+            if (isCalibrated()) {
+                sendMessage(new PowerInformationRequestMessage(macAddress));
+            }
+        }
+
+        @Override
+        public boolean shouldBeScheduled() {
+            return thing.getStatus() == ONLINE && (isLinked(CHANNEL_POWER)
+                    || configuration.getPowerStateChanging() != PowerStateChanging.COMMAND_SWITCHING);
+        }
+    };
+
+    private final PlugwiseDeviceTask energyUpdateTask = new PlugwiseDeviceTask("Energy update", scheduler) {
+        @Override
+        public int getConfiguredInterval() {
+            return getChannelUpdateInterval(CHANNEL_ENERGY);
+        }
+
+        @Override
+        public void runTask() {
+            if (isRecentLogAddressKnown()) {
+                updateEnergy();
+            }
+        }
+
+        @Override
+        public boolean shouldBeScheduled() {
+            return thing.getStatus() == ONLINE && isLinked(CHANNEL_ENERGY);
+        }
+    };
+
+    private final PlugwiseDeviceTask informationUpdateTask = new PlugwiseDeviceTask("Information update", scheduler) {
+        @Override
+        public int getConfiguredInterval() {
+            return Math.min(getChannelUpdateInterval(CHANNEL_STATE), getChannelUpdateInterval(CHANNEL_ENERGY));
+        }
+
+        @Override
+        public void runTask() {
+            updateInformation();
+        }
+
+        @Override
+        public boolean shouldBeScheduled() {
+            return thing.getStatus() == ONLINE && (isLinked(CHANNEL_STATE) || isLinked(CHANNEL_ENERGY));
+        }
+    };
+
+    private final PlugwiseDeviceTask realTimeClockUpdateTask = new PlugwiseDeviceTask("Real-time clock update",
+            scheduler) {
+        @Override
+        public int getConfiguredInterval() {
+            return getChannelUpdateInterval(CHANNEL_REAL_TIME_CLOCK);
+        }
+
+        @Override
+        public void runTask() {
+            sendMessage(new RealTimeClockGetRequestMessage(macAddress));
+        }
+
+        @Override
+        public boolean shouldBeScheduled() {
+            return thing.getStatus() == ONLINE && deviceType == DeviceType.CIRCLE_PLUS
+                    && isLinked(CHANNEL_REAL_TIME_CLOCK);
+        }
+    };
+
+    private final PlugwiseDeviceTask setClockTask = new PlugwiseDeviceTask("Set clock", scheduler) {
+        @Override
+        public int getConfiguredInterval() {
```

Comment:
The code has been reworked to make more use of the Java 8 Duration so exact units are more clear.